src/lib/EventListener/UserProfileListener.php

```diff
@@ -113,7 +113,9 @@ private function createRedirectToUserProfile(User $user): RedirectResponse
 
     private function isSupported(UserUpdateData $data): bool
     {
-        return $this->isUserProfileUpdate($data) && $this->canEditUserProfile($data->user);
+        return $this->isUserProfileUpdate($data) &&
+            $this->canEditUserProfile($data->user) &&
+            $this->doesOriginateFromProfileEditing();
```

Comment:
One more thing, actually.

Check route first. It's the least performance intensive operation. Permission checks should in my opinion be done last, as they can be costly.

Reply:
Basically we got back to the original solution, but this time code style suits your preferences more I think...

https://github.com/ibexa/admin-ui/pull/1143/commits/b52a34bcac03b8f7fce800cda4ad814a6c6533d0

Reply:
As it was supposed to be, but now the code is not duplicated and you can clearly see that both methods share it :sweat_smile: